src/lib/sec_profile.cpp

```diff
@@ -174,4 +175,16 @@ SecurityContext::s2k_iterations(pgp_hash_alg_t halg)
     return s2k_iterations_[halg];
 }
 
-} // namespace rnp
\ No newline at end of file
+void
+SecurityContext::set_time(uint64_t time) noexcept
+{
+    time_ = time;
+}
+
+uint64_t
+SecurityContext::time() const noexcept
+{
+    return time_ ? time_ : ::time(NULL);
```

Comment:
I am sure down the line somebody will want deterministic 1970-01-01 00:00:00 and will be frustrated :) Not that I'm worried too much about that, but maybe an extra "time-is-set" flag would keep some more people happy.

Reply:
@andrey-utkin We need to have a way to reset timestamp to the normal behaviour, so I used 0 value for this. Another way would be to add another function `rnp_unset_timestamp()`, but isn't that an overhead?

Reply:
@andrey-utkin any thoughts here? Thanks!